src/coordination/coordinator_instance.cpp

```diff
@@ -159,6 +164,122 @@ auto CoordinatorInstance::ShowInstances() const -> std::vector<InstanceStatus> {
   return instances_status;
 }
 
+void CoordinatorInstance::ForceResetCluster() {
+  // Force reset tries to return cluster to state in which we have all the replicas we had before
+  // and try to do failover to new MAIN. Only then is force reset successful
+  spdlog::info("Force resetting cluster!");
+  // Ordering is important here, we must stop frequent check before
+  // taking lock to avoid deadlock between us stopping thread and thread wanting to take lock but can't because
+  // we have it
+  std::ranges::for_each(repl_instances_, [](auto &repl_instance) { repl_instance.StopFrequentCheck(); });
+
+  auto lock = std::unique_lock{coord_instance_lock_};
+  repl_instances_.clear();
+  spdlog::info("Stopped all replication instance frequent checks.");
+
+  if (!raft_state_.IsLeader()) {
+    spdlog::trace("Exiting force reset as coordinator is not any more leader!");
+    return;
+  }
+
+  if (!raft_state_.AppendOpenLock()) {
+    spdlog::trace("Appending log force reset failed, aborting force reset");
+    return;
+  }
+
+  utils::OnScopeExit maybe_do_another_reset{[this]() {
+    if (raft_state_.IsLockOpened() && raft_state_.IsLeader()) {
+      spdlog::trace("Adding task to try force reset cluster again.");
+      thread_pool_.AddTask([this]() { this->ForceResetCluster(); });
+      return;
+    }
+    spdlog::trace("Lock is not opened anymore or coordinator is not leader, not doing force reset again.");
+  }};
+
+  auto const instances = raft_state_.GetReplicationInstances();
+
+  // To each instance we send RPC
+  // If RPC fails we consider instance dead
+  // Otherwise we consider instance alive
+  // If at any point later RPC fails for alive instance, we consider this failure
+
+  std::ranges::for_each(instances, [this](auto &replica) {
+    repl_instances_.emplace_back(this, replica.config, client_succ_cb_, client_fail_cb_,
+                                 &CoordinatorInstance::ReplicaSuccessCallback,
+                                 &CoordinatorInstance::ReplicaFailCallback);
+  });
+
+  auto const new_uuid = utils::UUID{};
+  auto const update_on_success_ping_on_swap_uuid = [&new_uuid](ReplicationInstance &instance) {
+    if (!instance.SendSwapAndUpdateUUID(new_uuid)) {
+      return;
+    }
+    instance.OnSuccessPing();
+  };
+  std::ranges::for_each(repl_instances_, update_on_success_ping_on_swap_uuid);
+
+  auto const is_swap_uuid_success = [](ReplicationInstance &instance) { return instance.IsAlive(); };
+
+  auto instances_with_swapped_uuid = repl_instances_ | ranges::views::filter(is_swap_uuid_success);
```

Comment:
Instead of swapping and then checking which was swapped you can apply general map principle with which you transform each replication instance into boolean based on the return value of SendSwapAndUpdateUUID

Reply:
I did it now, check if that was what you thought